cmd/cstor-pool-mgmt/pool/pool.go

```diff
@@ -0,0 +1,169 @@
+	}
+
+	importAttr = append(importAttr, "cstor-"+string(cStorPool.ObjectMeta.UID))
+
+	return importAttr
+}
+
+// CreatePool creates a new cStor pool.
+func CreatePool(cStorPool *apis.CStorPool) error {
+	createAttr := createPoolBuilder(cStorPool)
+	glog.V(4).Info("createAttr : ", createAttr)
+
+	stdoutStderr, err := RunnerVar.RunCombinedOutput(PoolOperator, createAttr...)
+	if err != nil {
+		glog.Error("Unable to create pool:", err.Error(), string(stdoutStderr))
+		return err
+	}
+
+	glog.Info("Creating Pool Successful")
+	return nil
+}
+
+// createPoolBuilder is to build create pool command.
+func createPoolBuilder(cStorPool *apis.CStorPool) []string {
+	// populate pool creation attributes.
+	var createAttr []string
+	createAttr = append(createAttr, "create")
+	if cStorPool.Spec.PoolSpec.CacheFile != "" {
+		cachefile := "cachefile=" + cStorPool.Spec.PoolSpec.CacheFile
+		createAttr = append(createAttr, "-o", cachefile)
+	}
+
+	openebsPoolname := "io.openebs:poolname=" + cStorPool.Name
+	createAttr = append(createAttr, "-O", openebsPoolname)
+
+	poolNameUID := "cstor-" + string(cStorPool.ObjectMeta.UID)
+	createAttr = append(createAttr, poolNameUID)
+
+	// To generate mirror disk0 disk1 mirror disk2 disk3 format.
+	for i, disk := range cStorPool.Spec.Disks.DiskList {
+		if cStorPool.Spec.PoolSpec.PoolType == "mirror" && i%3 == 0 {
+			createAttr = append(createAttr, "mirror")
+		}
+		createAttr = append(createAttr, disk)
+	}
+
+	return createAttr
+
+}
+
+// CheckValidPool checks for validity of CStorPool resource.
+func CheckValidPool(cStorPool *apis.CStorPool) error {
+	if string(cStorPool.ObjectMeta.UID) == "" {
+		return fmt.Errorf("Poolname cannot be empty")
+	}
+	if len(cStorPool.Spec.Disks.DiskList) < 1 {
+		return fmt.Errorf("Disk name(s) cannot be empty")
+	}
+	if cStorPool.Spec.PoolSpec.PoolType == "mirror" &&
+		len(cStorPool.Spec.Disks.DiskList)%2 != 0 {
+		return fmt.Errorf("Mirror poolType needs even number of disks")
+	}
+	return nil
+}
+
+// GetPoolName return the pool already created.
+func GetPoolName() (string, error) {
+	GetPoolStr := []string{"get", "-Hp", "name", "-o", "name"}
+	poolNameByte, err := RunnerVar.RunStdoutPipe(PoolOperator, GetPoolStr...)
+	if err != nil {
+		glog.Errorf("Unable to get pool:", poolNameByte)
+	}
+	poolName := string(poolNameByte)
+	glog.Infof("poolname : ", poolName)
+	return poolName, nil
+}
+
+// DeletePool destroys the pool created.
+func DeletePool(poolName string) error {
+	deletePoolStr := []string{"destroy", poolName}
+	stdoutStderr, err := RunnerVar.RunCombinedOutput(PoolOperator, deletePoolStr...)
+	if err != nil {
+		glog.Errorf("Unable to delete pool:", err.Error(), string(stdoutStderr))
+		return err
+	}
+	return nil
+}
+
+// CheckForZrepl is blocking call for checking status of zrepl in cstor-pool container.
+func CheckForZrepl() {
+	for {
+		_, err := RunnerVar.RunCombinedOutput(PoolOperator, "status")
+		if err != nil {
+			time.Sleep(3 * time.Second)
```

Comment:
Make 3 a constant and reduce the value to 1 second.

Reply:
will change. But why do we need to ping every 1 second?